core/src/main/java/com/graphhopper/routing/ev/IntEncodedValueImpl.java

```diff
@@ -243,10 +243,11 @@ public final int hashCode() {
 
     @Override
     public int getVersion() {
+        // todonow: there isn't really any 'version' here...
         int val = Helper.staticHashCode(name);
         val = 31 * val + (storeTwoDirections ? 1231 : 1237);
         val = 31 * val + (negateReverseDirection ? 13 : 17);
-        return staticHashCode(val, fwdDataIndex, bwdDataIndex, bits, minValue, maxValue, fwdShift, bwdShift, fwdMask, bwdMask);
+        return staticHashCode(val, bits, minValue, maxValue);
```

Comment:
I wasn't entirely sure how to treat these fields. They represent the position of the encoded values in the edge flags (I think), so they are only really useful in the context of the encoding manager. So far I excluded them from the serialization string. I check the version when deserializing the encoded values (see EncodedValueSerializer), but since the encoded values are not added to an encoding manager at this point these fields are all zero, i.e. they do not match the values that were used during serialization.

We have two options: Either include all these fields in the serialization string and in the version, or exclude them in both cases. Or maybe a third option: Skip the version check when deserializing the encoded values, which also makes kind of sense at least as long `getVersion()` does not really include a 'version' (see above comment).
For now I decided to exclude these fields in both cases, because I thought they do not really belong to the 'identity' of a single encoded value. Btw: Why do we need non-default equals/hashCode implementations here?

Reply:
Ok, I am not entirely sure why I left out these fields in https://github.com/graphhopper/graphhopper/pull/2542. I found this comment:

https://github.com/graphhopper/graphhopper/blob/acd82a80784a20a2c51626609347447fa1fb5bb5/core/src/main/java/com/graphhopper/routing/ev/IntEncodedValueImpl.java#L48-L49

So probably I ran into the same problem as now: As long as we had no way of adding encoded values to the encoding manager without having the encoding manager set these fields automatically it was easiest to leave them out during serialization. But now this should be possible by adding an extra method to `EncodingManager#Builder`.

Reply:
They are included now, so now we store the complete state of the EM in the properties file and can later load it again :+1: